### PR TITLE
refactor: add metadata render interface and apply it to `oras attach` command

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #1573 

**Please check the following list**:
- [ ]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

File: cmd/oras/internal/display/handler.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func NewPushHandler(printer *output.Printer, format option.Format, tty *os.File,
 }
 
 // NewAttachHandler returns status and metadata handlers for attach command.
-func NewAttachHandler(printer *output.Printer, format option.Format, tty *os.File, fetcher fetcher.Fetcher) (status.AttachHandler, metadata.AttachHandler, error) {
+func NewAttachHandler(printer *output.Printer, format option.Format, tty *os.File, fetcher fetcher.Fetcher, target *option.Target) (status.AttachHandler, metadata.AttachHandler, error) {
```
Comment (qweeah) [inline cmd/oras/internal/display/handler.go:65]: Adding `target *option.Target` as a parameter of `OnAttached` would incur less change

File: cmd/oras/internal/display/metadata/interface.go [hunk 2]
```diff
@@ -28,9 +28,15 @@ type PushHandler interface {
 	OnCompleted(root ocispec.Descriptor) error
 }
 
+// Renderer renders metadata information when an operation is complete.
+type Renderer interface {
+	OnCompleted() error
```
Comment (qweeah) [inline cmd/oras/internal/display/metadata/interface.go:33]:
```suggestion
	Render() error
```

Comment (wangxiaoxuan273) [reply cmd/oras/internal/display/metadata/interface.go:33]: I think `OnCompleted` is a better name. `Render` is confusing and users wouldn't know what it does.

**Review by qweeah — APPROVED** [review]
LGTM

File: cmd/oras/internal/display/metadata/interface.go (hunk 2 above)
Comment (qweeah) [reply cmd/oras/internal/display/metadata/interface.go]: Combining with the package name, `metadata.Render` should provide enough information on what was being operated on.

**Pull request merged**